src/key-state.c

```diff
@@ -103,8 +103,38 @@ key_state_nr_bound_keys(void)
 	return bound.nr_keys;
 }
 
-int
-key_state_nr_pressed_keys(void)
+/*
+ * Reference:
+ * https://github.com/xkbcommon/libxkbcommon/blob/HEAD/include/xkbcommon/xkbcommon-keysyms.h
+ */
+bool
+key_state_multiple_normal_keys_pressed(void)
 {
-	return pressed.nr_keys;
+	if (pressed.nr_keys < 2) {
+		return false;
+	}
+
+	/*
+	 * If the pressed array contains multiple keys, we have to analyze a bit
+	 * further because some keys like those listed below do not always
+	 * receive release events.
+	 *   - XKB_KEY_ISO_Group_Next
```

Comment:
The key that sometimes is identified as `XKB_KEY_ISO_Group_Next` should always receive a release event.
This should be fixed by always removing the key on release events.